containerized_ml_models/model_api_14/api.py

```diff
@@ -0,0 +1,25 @@
+from typing import Any
+from flask import Flask, request, jsonify
+import numpy as np
+import pickle
+import ast
+
+app = Flask(__name__)
+with open("model.pkl", "rb") as file:
+    model = pickle.load(file)
```

Comment:
I like the fact that you load the image outside of the predict function. However, I would put it into a singleton class' object  not to make it globally accessible ;)

Reply:
Where do you propose to store this singleton class object? If as a global variable, what is the purpose of storing model in singleton class object?

Reply:
What about storing a model in a private instance field? Loaded while object creation it cannot be modified later.

Reply:
In the way it was done in #6?

Reply:
Forgive me for digging into the subject, but I've often heard the term that singleton is anti-pattern.
You can read its disadvantages here https://refactoring.guru/pl/design-patterns/singleton

Although as I read about it, it would actually be applicable to our small project. However, for the future it could make some problems in a larger project.

Reply:
If I understand the code in #6 `self.model` is not a private field, but I mean this kind of encapsulation.

I would additionally change the field name to self.__model so you cannot call <instance>.model and modify it

Reply:
> Forgive me for digging into the subject, but I've often heard the term that singleton is anti-pattern. You can read its disadvantages here https://refactoring.guru/pl/design-patterns/singleton
> 
> Although as I read about it, it would actually be applicable to our small project. However, for the future it could make some problems in a larger project.

Yeah, you know, in the future, we would rather use mlflow or a similar tool for saving and serving